pkg/jsonfilter/jsonfilter.go

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright 2025 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jsonfilter
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	entsql "entgo.io/ent/dialect/sql"
+	"entgo.io/ent/dialect/sql/sqljson"
+)
+
+// columnNotFoundSQLState is the PostgresSQL state for column not found error.
+// Reference: https://www.postgresql.org/docs/8.2/errcodes-appendix.html
+const columnNotFoundSQLState = "42703"
+
+// columnNotExistRegex extracts column name from the error message
+var columnNotExistRegex = regexp.MustCompile(`column "([^"]+)" does not exist`)
```

Comment:
I'd not support dynamic selection of columns, that's an implementation detail to me that should not be exposed in then API